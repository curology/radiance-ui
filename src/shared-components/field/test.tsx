import React from 'react';
import { render } from 'src/tests/testingLibraryHelpers';

import { Field } from './index';

describe('<Field />', () => {
  /**
   * TODO: Fix Emotion 11 CI snapshot serializer order issue
   */
  // eslint-disable-next-line jest/no-disabled-tests
  describe.skip('UI snapshots', () => {
    it('renders with default props', () => {
      const { container } = render(
        <Field>
          <Field.Input />
        </Field>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('renders with label and labelFor props', () => {
      const labelText = 'Test Label';
      const labelFor = 'for-input-id';
      const { container } = render(
        <Field label={labelText} labelFor={labelFor}>
          <Field.Input />
        </Field>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });

    it('renders with errorMessage, hintMessage and hideMessagesIcon props', () => {
      const errorMessage = (
        <React.Fragment>
          <strong>Uh Oh!</strong> Type again
        </React.Fragment>
      );
      const { container } = render(
        <Field
          hideMessagesIcon
          hintMessage="hint message"
          messages={{ error: errorMessage }}
        >
          <Field.Input />
        </Field>,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });

  describe('when provided some messages', () => {
    it('renders the error message', () => {
      const maxLength = 'Maximum 6 characters';
      const messages = { maxLength };
      const { getAllByText } = render(
        <Field messages={messages} messagesType="error">
          <Field.Input />
        </Field>,
      );

      const message = getAllByText(maxLength);
      expect(message).toHaveLength(1);
    });

    it('renders the success messages', () => {
      const success = 'Thanks for completing';
      const messages = { success: 'Thanks for completing' };
      const { getAllByText } = render(
        <Field messages={messages} messagesType="success">
          <Field.Input />
        </Field>,
      );

      const message = getAllByText(success);
      expect(message).toHaveLength(1);
    });
  });
});
