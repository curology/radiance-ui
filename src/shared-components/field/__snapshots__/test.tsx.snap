// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> UI snapshots renders with default props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 22px;
}

.emotion-2 {
  position: relative;
}

.emotion-2 svg.radiance-field-input-icon {
  opacity: 0;
  position: absolute;
  top: 20px;
  right: 1rem;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
}

.emotion-4 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #BD200F;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-5 {
  color: #524D6E;
  font-size: 1rem;
  line-height: 1.75;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: #FFFFFF;
  border: 1px solid #EDEDF0;
  border-radius: 4px;
  color: #524D6E;
  -webkit-transition: border-color 350ms;
  transition: border-color 350ms;
  width: 100%;
  outline: none;
  margin-bottom: 2px;
  padding: 13px 2rem 13px 1rem;
}

.emotion-5:active,
.emotion-5:focus {
  outline: none;
  border-color: #332E54;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-5:active~.e1gqjikl5,
.emotion-5:focus~.e1gqjikl5 {
  max-height: 24px;
  opacity: 1;
  margin: 0.125rem 0 0 0;
}

.emotion-5::-webkit-input-placeholder {
  color: #C3C0CD;
}

.emotion-5::-moz-placeholder {
  color: #C3C0CD;
}

.emotion-5:-ms-input-placeholder {
  color: #C3C0CD;
}

.emotion-5::placeholder {
  color: #C3C0CD;
}

.emotion-5[disabled] {
  background-color: #EDEDF0;
  color: #C3C0CD;
  cursor: not-allowed;
  border: 1px solid #EDEDF0;
  box-shadow: none;
}

.emotion-7 {
  list-style-type: none;
  margin: 0;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <svg
      class="radiance-field-input-icon emotion-4"
      fill="#BD200F"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M14.6 6.79a6.747 6.747 0 0 0-7.9-5.36 6.7 6.7 0 0 0-4.31 2.84 6.71 6.71 0 0 0-1.04 5.06 6.747 6.747 0 0 0 7.9 5.36 6.7 6.7 0 0 0 4.31-2.84 6.71 6.71 0 0 0 1.04-5.06ZM6.46.2C6.97.11 7.48.06 7.98.06a7.996 7.996 0 0 1 1.5 15.85c-.51.09-1.02.14-1.52.14-3.76 0-7.12-2.66-7.85-6.49A7.996 7.996 0 0 1 6.46.2Zm.86 3.91h1.41v5.26H7.32V4.11Zm-.17 6.83c0-.48.39-.84.88-.84s.88.36.88.84c0 .49-.39.86-.88.86s-.88-.37-.88-.86Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <input
      class="emotion-5 emotion-6"
    />
    <ul
      class="emotion-7 emotion-8"
    />
  </div>
</div>
`;

exports[`<Field /> UI snapshots renders with errorMessage, hintMessage and hideMessagesIcon props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 22px;
}

.emotion-2 {
  position: relative;
}

.emotion-2 svg.radiance-field-input-icon {
  opacity: 0;
  position: absolute;
  top: 20px;
  right: 1rem;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
}

.emotion-2 svg.radiance-field-input-icon {
  opacity: 1;
}

.emotion-2 .e1gqjikl1,
.emotion-2 .emotion-5 {
  border-color: #BD200F;
}

.emotion-2 .e1gqjikl1:active,
.emotion-2 .emotion-5:active,
.emotion-2 .e1gqjikl1:focus,
.emotion-2 .emotion-5:focus {
  border-color: #BD200F;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-4 {
  color: #524D6E;
  font-size: 1rem;
  line-height: 1.75;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: #FFFFFF;
  border: 1px solid #EDEDF0;
  border-radius: 4px;
  color: #524D6E;
  -webkit-transition: border-color 350ms;
  transition: border-color 350ms;
  width: 100%;
  outline: none;
  margin-bottom: 2px;
  padding: 13px 2rem 13px 1rem;
}

.emotion-4:active,
.emotion-4:focus {
  outline: none;
  border-color: #332E54;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-4:active~.emotion-7,
.emotion-4:focus~.emotion-7 {
  max-height: 24px;
  opacity: 1;
  margin: 0.125rem 0 0 0;
}

.emotion-4::-webkit-input-placeholder {
  color: #C3C0CD;
}

.emotion-4::-moz-placeholder {
  color: #C3C0CD;
}

.emotion-4:-ms-input-placeholder {
  color: #C3C0CD;
}

.emotion-4::placeholder {
  color: #C3C0CD;
}

.emotion-4[disabled] {
  background-color: #EDEDF0;
  color: #C3C0CD;
  cursor: not-allowed;
  border: 1px solid #EDEDF0;
  box-shadow: none;
}

.emotion-6 {
  color: #706D87;
  font-size: 0.875rem;
  line-height: 1.71;
  -webkit-transition: all 350ms ease-in-out;
  transition: all 350ms ease-in-out;
  opacity: 0;
  max-height: 0;
}

.emotion-8 {
  list-style-type: none;
  margin: 0;
}

.emotion-10 {
  color: #706D87;
  font-size: 0.875rem;
  line-height: 1.71;
  color: #BD200F;
  line-height: 24px;
}

.emotion-10:last-of-type {
  margin: 0 0 0.125rem 0;
}

<div
  class="emotion-0 emotion-1"
>
  <div
    class="emotion-2 emotion-3"
  >
    <input
      aria-invalid="true"
      class="emotion-4 emotion-5"
    />
    <div
      class="emotion-6 emotion-7"
    >
      hint message
    </div>
    <ul
      class="emotion-8 emotion-9"
    >
      <li
        class="emotion-10 emotion-11"
        style="opacity: 1; max-height: 48px; transition-duration: 350ms; transition-property: max-height, opacity; transition-timing-function: ease-in-out;"
        type="error"
      >
        <output
          role="alert"
        >
          <strong>
            Uh Oh!
          </strong>
           Type again
        </output>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<Field /> UI snapshots renders with label and labelFor props 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 22px;
}

.emotion-2 {
  color: #524D6E;
  font-size: 0.75rem;
  line-height: 1.67;
}

.emotion-4 {
  position: relative;
}

.emotion-4 svg.radiance-field-input-icon {
  opacity: 0;
  position: absolute;
  top: 20px;
  right: 1rem;
  -webkit-transition: opacity 350ms;
  transition: opacity 350ms;
}

.emotion-6 {
  display: block;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #BD200F;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  transition: color 350ms,transform 350ms;
}

.emotion-7 {
  color: #524D6E;
  font-size: 1rem;
  line-height: 1.75;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  background: #FFFFFF;
  border: 1px solid #EDEDF0;
  border-radius: 4px;
  color: #524D6E;
  -webkit-transition: border-color 350ms;
  transition: border-color 350ms;
  width: 100%;
  outline: none;
  margin-bottom: 2px;
  padding: 13px 2rem 13px 1rem;
}

.emotion-7:active,
.emotion-7:focus {
  outline: none;
  border-color: #332E54;
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
}

.emotion-7:active~.e1gqjikl5,
.emotion-7:focus~.e1gqjikl5 {
  max-height: 24px;
  opacity: 1;
  margin: 0.125rem 0 0 0;
}

.emotion-7::-webkit-input-placeholder {
  color: #C3C0CD;
}

.emotion-7::-moz-placeholder {
  color: #C3C0CD;
}

.emotion-7:-ms-input-placeholder {
  color: #C3C0CD;
}

.emotion-7::placeholder {
  color: #C3C0CD;
}

.emotion-7[disabled] {
  background-color: #EDEDF0;
  color: #C3C0CD;
  cursor: not-allowed;
  border: 1px solid #EDEDF0;
  box-shadow: none;
}

.emotion-9 {
  list-style-type: none;
  margin: 0;
}

<div
  class="emotion-0 emotion-1"
>
  <label
    class="emotion-2 emotion-3"
    for="for-input-id"
  >
    Test Label
  </label>
  <div
    class="emotion-4 emotion-5"
  >
    <svg
      class="radiance-field-input-icon emotion-6"
      fill="#BD200F"
      height="16"
      viewBox="0 0 16 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M14.6 6.79a6.747 6.747 0 0 0-7.9-5.36 6.7 6.7 0 0 0-4.31 2.84 6.71 6.71 0 0 0-1.04 5.06 6.747 6.747 0 0 0 7.9 5.36 6.7 6.7 0 0 0 4.31-2.84 6.71 6.71 0 0 0 1.04-5.06ZM6.46.2C6.97.11 7.48.06 7.98.06a7.996 7.996 0 0 1 1.5 15.85c-.51.09-1.02.14-1.52.14-3.76 0-7.12-2.66-7.85-6.49A7.996 7.996 0 0 1 6.46.2Zm.86 3.91h1.41v5.26H7.32V4.11Zm-.17 6.83c0-.48.39-.84.88-.84s.88.36.88.84c0 .49-.39.86-.88.86s-.88-.37-.88-.86Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
    <input
      class="emotion-7 emotion-8"
    />
    <ul
      class="emotion-9 emotion-10"
    />
  </div>
</div>
`;
